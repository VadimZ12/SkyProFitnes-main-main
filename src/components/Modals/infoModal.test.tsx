import React from 'react'
import { render } from '@testing-library/react'
import InfoModal from './infoModal'

describe('InfoModal Snapshot', () => {
	it('renders progress modal correctly', () => {
		const { asFragment } = render(
			<React.Fragment>
			<InfoModal
				isOpen={true}
				onClose={() => {}}
				message='Operation in progress'
				type='progress'
			/>
			</React.Fragment>
		)
		expect(asFragment()).toMatchSnapshot()
	})

	it('renders reset password modal correctly', () => {
		const { asFragment } = render(
			<React.Fragment>
			<InfoModal
				isOpen={true}
				onClose={() => {}}
				message='Check your email'
				type='resetPassword'
				email='test@example.com'
			/>
			</React.Fragment>
		)
		expect(asFragment()).toMatchSnapshot()
	})

	it('renders nothing when modal is closed', () => {
		const { asFragment } = render(
			<React.Fragment>
			<InfoModal
				isOpen={false}
				onClose={() => {}}
				message='Operation in progress'
				type='progress'
			/>
			</React.Fragment>
		)
		expect(asFragment()).toMatchSnapshot()
	})
})
