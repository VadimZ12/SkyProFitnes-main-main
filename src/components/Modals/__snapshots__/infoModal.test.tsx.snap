// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModal Snapshot renders nothing when modal is closed 1`] = `<DocumentFragment />`;

exports[`InfoModal Snapshot renders progress modal correctly 1`] = `
<DocumentFragment>
  <div
    class="
    fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center z-50
  "
  >
    <div
      class="
    bg-white flex flex-col items-center rounded-[30px]
    shadow-[0px_4px_67px_-12px_rgba(0,0,0,0.13)]
    w-[444px] h-[360px] p-[33px] gap-[34px]
"
    >
      <p
        class="
    font-roboto text-[40px] font-[450] leading-[48px] text-center mb-[30px]
  "
      >
        Operation in progress
      </p>
      <svg
        class="
    w-[68px] h-[68px] text-[#BCEC30]
  "
        fill="none"
        viewBox="0 0 68 68"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34 0C15.2452 0 0 15.2452 0 34C0 52.7548 15.2452 68 34 68C52.7548 68 68 52.7548 68 34C68 15.2452 52.7548 0 34 0ZM27.2 51L10.2 34L15.1294 29.206L27.2 41.1378L52.8706 15.5668L57.8 20.4L27.2 51Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InfoModal Snapshot renders reset password modal correctly 1`] = `
<DocumentFragment>
  <div
    class="
    fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center z-50
  "
  >
    <div
      class="
    bg-white flex flex-col items-center rounded-[30px]
    shadow-[0px_4px_67px_-12px_rgba(0,0,0,0.13)]
    w-[360px] h-[223px] p-[40px] gap-[48px]
"
    >
      <img
        alt="SkyFitnessPro"
        class="w-[220px] h-[35px]"
        loading="lazy"
        src="/logo.png"
      />
      <p
        class="text-center font-roboto text-[18px] leading-[22px]"
      >
        Ссылка для восстановления
        <br />
        пароля отправлена
        <br />
        на test@example.com
      </p>
    </div>
  </div>
</DocumentFragment>
`;
